File: backend-asp-school/backend-asp-school/Models/Student.cs
using System.Security.Claims;
using System.Text.Json.Serialization;

namespace backend_asp_school.Models
{
    public class Student
    {
        public int StudentID { get; set; }
        public string StudentName { get; set; } = null!;
        public DateTime? BirthDate { get; set; }
        public string? Gender { get; set; }

        public int? ClassID { get; set; }
      
        public Classes? Class { get; set; }

        public ICollection<Result>? Results { get; set; }
        public ICollection<Parent>? Parents { get; set; }
    }
}

File: next-dashboard-ui/src/lib/student.ts
import { Class } from "./class";
import { Result } from "./result";
import { Parent } from "./parent";

export interface Student {
  studentID: number;
  studentName: string;
  birthDate?: string;   // ISO string (YYYY-MM-DD)
  gender?: string;

  classID?: number;
  class?: Class;

  results?: Result[];
  parents?: Parent[];
}
